samples/apply-stretch-renderer/src/main/java/com/esri/arcgismaps/sample/applystretchrenderer/components/ApplyStretchRendererViewModel.kt

```diff
@@ -0,0 +1,260 @@
+    private var appliedMinValue = DEFAULT_MIN
+    private var appliedMaxValue = DEFAULT_MAX
+    private var appliedPercentMin = DEFAULT_PERCENT_MIN
+    private var appliedPercentMax = DEFAULT_PERCENT_MAX
+    private var appliedStdDeviationFactor = DEFAULT_STD_DEVIATION_FACTOR
+    private var appliedStretchType = StretchType.MinMax
+
+    // The map used in the sample, with imagery basemap
+    val arcGISMap: ArcGISMap = ArcGISMap(BasemapStyle.ArcGISImageryStandard)
+
+    // MapViewProxy used to set viewpoint after layer loads
+    val mapViewProxy = MapViewProxy()
+
+    // Message dialog view model for error handling
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Provision path for local offline resources (raster-file/Shasta.tif)
+    private val provisionPath: String by lazy {
+        app.getExternalFilesDir(null)?.path.toString() + File.separator +
+                app.getString(R.string.apply_stretch_renderer_app_name)
+    }
+
+    // The raster data (Shasta.tif) should be downloaded to external storage on launch
+    private val raster: Raster by lazy {
+        val rasterFile = File(provisionPath, "raster-file${File.separator}Shasta.tif")
+        require(rasterFile.exists()) { "Invalid raster data: file does not exist at ${rasterFile.path}" }
+        Raster.createWithPath(rasterFile.path)
+    }
+
+    // The raster layer to which the stretch renderer will be applied
+    private val rasterLayer: RasterLayer = RasterLayer(raster)
+
+    // Stretch type options for UI
+    val stretchTypeOptions: List<String> = listOf("MinMax", "Percent Clip", "Std Deviation")
+
+    // Current selected stretch type
+    private val _selectedStretchType = MutableStateFlow(appliedStretchType)
+    val selectedStretchType = _selectedStretchType.asStateFlow()
+
+    // Min-Max parameters (values represent pixel value range)
+    private val _minValue = MutableStateFlow(DEFAULT_MIN)
+    val minValue = _minValue.asStateFlow()
+
+    private val _maxValue = MutableStateFlow(DEFAULT_MAX)
+    val maxValue = _maxValue.asStateFlow()
+
+    // Percent clip parameters (values represent percent 0..100)
+    private val _percentMin = MutableStateFlow(DEFAULT_PERCENT_MIN)
+    val percentMin = _percentMin.asStateFlow()
+
+    private val _percentMax = MutableStateFlow(DEFAULT_PERCENT_MAX)
+    val percentMax = _percentMax.asStateFlow()
+
+    // Standard deviation factor (typical range 0.25..4.0)
+    private val _stdDeviationFactor = MutableStateFlow(DEFAULT_STD_DEVIATION_FACTOR)
+    val stdDeviationFactor = _stdDeviationFactor.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            rasterLayer.load().onSuccess {
+                arcGISMap.operationalLayers.add(rasterLayer)
+            }.onFailure {
+                messageDialogVM.showMessageDialog(it)
+            }
+
+            arcGISMap.load().onFailure {
+                messageDialogVM.showMessageDialog(it)
```

Comment:
Same as above, return if the map fails to load: 
```suggestion
                return@launch messageDialogVM.showMessageDialog(it)
```